crates/accelerate/src/convert_2q_block_matrix.rs

```diff
@@ -32,32 +38,63 @@ pub fn blocks_to_matrix(
     py: Python,
     op_list: Vec<(PyReadonlyArray2<Complex64>, SmallVec<[u8; 2]>)>,
 ) -> PyResult<Py<PyArray2<Complex64>>> {
-    let identity = aview2(&ONE_QUBIT_IDENTITY);
-    let input_matrix = op_list[0].0.as_array();
-    let mut matrix: Array2<Complex64> = match op_list[0].1.as_slice() {
-        [0] => kron(&identity, &input_matrix),
-        [1] => kron(&input_matrix, &identity),
-        [0, 1] => input_matrix.to_owned(),
-        [1, 0] => change_basis(input_matrix),
-        [] => Array2::eye(4),
+    let identity = aview2(&ONE_QUBIT_IDENTITY).into_faer_complex();
+    let input_matrix = op_list[0].0.as_array().into_faer_complex();
+    let mut matrix: Mat<c64> = Mat::<c64>::zeros(4, 4);
+
+    match op_list[0].1.as_slice() {
+        [0] => kron(matrix.as_mut(), identity.as_ref(), input_matrix.as_ref()),
+        [1] => kron(matrix.as_mut(), input_matrix.as_ref(), identity.as_ref()),
+        [0, 1] => matrix = input_matrix.to_owned(),
+        [1, 0] => matrix = change_basis_faer(input_matrix.to_owned()),
+        [] => matrix = Mat::<c64>::identity(4, 4),
         _ => unreachable!(),
     };
     for (op_matrix, q_list) in op_list.into_iter().skip(1) {
-        let op_matrix = op_matrix.as_array();
+        let op_matrix = op_matrix.as_array().into_faer_complex();
+        let mut result: Mat<c64> = Mat::<c64>::zeros(4, 4);
 
-        let result = match q_list.as_slice() {
-            [0] => Some(kron(&identity, &op_matrix)),
-            [1] => Some(kron(&op_matrix, &identity)),
-            [1, 0] => Some(change_basis(op_matrix)),
-            [] => Some(Array2::eye(4)),
-            _ => None,
-        };
-        matrix = match result {
-            Some(result) => result.dot(&matrix),
-            None => op_matrix.dot(&matrix),
+        match q_list.as_slice() {
+            [0] => kron(result.as_mut(), identity.as_ref(), op_matrix.as_ref()),
+            [1] => kron(result.as_mut(), op_matrix.as_ref(), identity.as_ref()),
+            [1, 0] => result = change_basis_faer(op_matrix.to_owned()),
+            [] => result = Mat::<c64>::identity(4, 4),
+            _ => result = op_matrix.to_owned(),
         };
+
+        let aux = matrix.clone();
+        matmul(
+            matrix.as_mut(),
+            result.as_ref(),
+            aux.as_ref(),
+            None,
+            c64::new(1., 0.),
+            Parallelism::None,
+        );
     }
-    Ok(matrix.into_pyarray_bound(py).unbind())
+
+    Ok(matrix
+        .as_ref()
+        .into_ndarray_complex()
+        .to_owned()
+        .into_pyarray_bound(py)
+        .unbind())
+}
+
+/// Switches the order of qubits in a two qubit operation.
+/// This function will substitue `change_basis` once the
+/// `two_qubit_decompose.rs` uses Mat<c64> instead of ArrayView2
+#[inline]
+pub fn change_basis_faer(matrix: Mat<c64>) -> Mat<c64> {
```

Comment:
The idea is that this function ends up replacing `change_basis` once the `two_qubit_decompose.rs` uses `Mat<c64>` instead of `ArrayView2`. I created another function just to avoid modifying the other file until the follow-up. After that, the function will be renamed to `change_basis` again.

Reply:
Arnau and I discussed this as the best approach because otherwise two_qubit_decompose.rs needed non-trivial edits.